app/assets/javascripts/admin/bulk_product_update.js.coffee

```diff
@@ -212,10 +212,10 @@ angular.module("ofn.admin").controller "AdminProductEditCtrl", ($scope, $timeout
       ).error (data, status) ->
         if status == 400 && data.errors? && data.errors.length > 0
           errors = error + "\n" for error in data.errors
-          alert "Saving failed with the following error(s):\n" + errors
-          $scope.displayFailure "Save failed due to invalid data"
+          alert t("products_update_error") + errors
```

Comment:
Test fails with this (note line break after the : in the expected string):
```
Expected spy alert to have been called with [ 'Saving failed with the following error(s):

	an error

	' ] but actual calls were [ 'Saving failed with the following error(s):an error

	' ]
```

Your translated string replaced `alert "Saving failed with the following error(s):\n" + errors` but you are missing the newline \n. So I think should be something like:

```{coffeescript}
alert t("products_update_error") + "\n" + errors
```

I think it would be better to keep punctuation in the view and only translate the words as far as possible (unless a decision has been made to keep all translations unique to their context on the site?). Then the translations can be reused elsewhere where it might not be appropriate to have them followed by a colon (for example).